File: daaexam/dfs.java


public class dfs{
    public static void main(String[] args) {
        int[][] graph = {
            {1, 2},    // neighbors of 0
            {0, 3},    // neighbors of 1
            {0, 4},    // neighbors of 2
            {1},       // neighbors of 3
            {2}        // neighbors of 4
        };

        boolean[] visited = new boolean[5];
        System.out.println("DFS:");
        dfs(0, graph, visited);
    }

    static void dfs(int node, int[][] graph, boolean[] visited) {
        visited[node] = true;
        System.out.print(node + " ");

        for (int neighbor : graph[node]) {
            if (!visited[neighbor])
                dfs(neighbor, graph, visited);
        }
    }
}

File: daaexam/dfs.py
graph = {
    'A': ['B', 'C'],
    'B': ['D', 'E'],
    'C': ['F'],
    'D': [],
    'E': [],
    'F': []
}

def dfs(graph, node, visited=[]):
    if node not in visited:
        print(node, end=' ')
        visited.append(node)
        for neighbor in graph[node]:
            dfs(graph, neighbor, visited)

print("DFS:")
dfs(graph, 'A')